keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,646 @@
+                    padding="same",
+                    activation="softmax",
+                )
+
+                input_shape = input_tensor.shape
+
+                x = backbone(input_tensor)
+
+                output_shape = x.shape
+
+                if input_shape[0] is None and input_shape[1] is None:
+                    if None in backbone.input.shape:
+                        raise ValueError(
+                            f"""No `input_shape` argument available to model. `input_shape`
+                            received by `FullyConvolutionalNetwork` is {input_shape} and
+                            `input_shape` passed to backbone is {backbone.input.shape}"""
+                        )
+                    else:
+                        input_shape = backbone.input.shape
+
+                target_height_factor = input_shape[1] // output_shape[1]
+                target_width_factor = input_shape[2] // output_shape[2]
+
+                data_format = tf.keras.backend.image_data_format()
+
+                upscale = tf.keras.layers.UpSampling2D(
+                    size=(target_height_factor, target_width_factor),
+                    data_format=data_format,
+                    interpolation="bilinear",
+                )
+
+                x = classes_conv(x)
+                output_tensor = upscale(x)
+
+                if return_mask:
+                    if data_format == "channels_last":
+                        output_tensor = tf.math.argmax(output_tensor, axis=3)
+                        output_tensor = tf.expand_dims(output_tensor, axis=3)
+                    else:
+                        output_tensor = tf.math.argmax(output_tensor, axis=1)
+                        output_tensor = tf.expand_dims(output_tensor, axis=1)
+
+                output_tensor = tf.cast(output_tensor, tf.float32)
+
+                super().__init__(
+                    inputs={"input_tensor": input_tensor},
+                    outputs={"output_tensor": output_tensor},
+                )
+
+                self.classes = classes
+                self.model_architecture = model_architecture
+                self.backbone = backbone
+                self.return_mask = return_mask
+        else:
+            raise ValueError(
+                "Invalid argument for parameter `backbone`. Accepted values are a `keras_cv.models.*`, `tf.keras.models.Model` or pre-supported `keras_cv.models.VGG16`/`keras_cv.models.VGG19`"
+            )
+
+    def get_config(self):
+        config = {
+            "classes": self.classes,
+            "backbone": self.backbone,
+            "model_architecture": self.model_architecture,
+            "return_mask": self.return_mask,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+def FCN8S(
+    classes, backbone, return_mask=False, input_shape=(None, None, 3), input_tensor=None
+):
+    if input_shape == (None, None, 3):
```

Comment:
You can have this check once in the FullyConvolutionalNetwork instead of in each method?

Reply:
Can do, yes. Will change it